### PR TITLE
Solution for Chapter One

### PR DESCRIPTION
**Pull request by mcartaud**
### Solutions for Chapter One

cc @vrom911 @chshersh

Sorry for the deleted pull request (and notifications) on the main repo, I didn't use pull request usually.
Thanks again for your time on the reviews and for the challenge :)

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -542,7 +545,11 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+  | (abs x < abs y) && (abs y < abs z) || (abs z < abs y) && (abs y < abs x) = y
+  | (abs x < abs z) && (abs z < abs y) || (abs y < abs z) && (abs z < abs x) = z
```
Comment (chshersh) [inline src/Chapter1.hs]: I wonder if `abs` can lead to wrong behaviour in some situations :thinking: 

For example, if `x = 1, y = -2, z = 3`, this function returns `-2`, while the correct answer is `1`.

If you put those numbers in the increasing order, you will get `-2 1 3` and that's why `1` is in the middle.

File: src/Chapter1.hs [hunk 2]
```diff
@@ -620,7 +634,11 @@ Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n =
+  let theLast = lastDigit n
+      theLastButOne = lastDigit (div (abs n) 10)
+   in theLast + theLastButOne
```
Comment (chshersh) [inline src/Chapter1.hs:641]: Very nice work on reusing the `lastDigit` function from the previous task! :clap:

**Review by chshersh — APPROVED** [review]
Nice solutions! Congratulations with finishing Chapter One :tada: 
And bonus points for cracking the Advanced Task :medal_sports: 

I left my feedback as a review :slightly_smiling_face: 

And no worries about the PR in the original repo :ok:

**Comment by chshersh** [comment]
⚠️ By the way, per new rules of the Hacktoberfest event, you would need to add the `hacktoberfest` topic to this fork repo:

* https://hacktoberfest.digitalocean.com/hacktoberfest-update

All previous contributions are legit, but from now on it should have the label to participate.

File: src/Chapter1.hs (hunk 1 above)
Comment (mcartaud) [reply src/Chapter1.hs]: Yes, I can't recall why I put them in the first place -_-'.
Thanks for pointing it out

**Comment by chshersh** [comment]
@mcartaud Well done! I think this PR is ready for merge :checkered_flag:

**Pull request merged**